common/ayon_common/distribution/addon_distribution.py

```diff
@@ -294,203 +343,710 @@ def get_dependency_package(package_name=None):
+        package = self.dependency_package
+        if package is None or not package.require_distribution:
+            return None
+
+        metadata = self.get_dependency_metadata()
+        downloader_data = {
+            "type": "dependency_package",
+            "name": package.name,
+            "platform": package.platform
+        }
+        zip_dir = package_dir = os.path.join(
+            self._dependency_dirpath, package.name
+        )
+        self.log.debug(f"Checking {package.name} in {package_dir}")
+
+        if not os.path.isdir(package_dir) or package.name not in metadata:
+            state = UpdateState.OUTDATED
+        else:
+            state = UpdateState.UPDATED
+
+        return DistributionItem(
+            state,
+            zip_dir,
+            package_dir,
+            package.checksum,
+            self._dist_factory,
+            package.sources,
+            downloader_data,
+            package.name,
+            self.log,
+        )
 
-    if log is None:
-        log = logging.getLogger(__name__)
+    def get_addons_dist_items(self):
+        if self._addons_dist_items is None:
+            self._addons_dist_items = self._prepare_current_addons_dist_items()
+        return self._addons_dist_items
 
-    package = get_dependency_package()
-    if not package:
-        log.info("Server does not contain dependency package")
-        return
+    def get_dependency_dist_item(self):
+        if self._dependency_dist_item == -1:
+            self._dependency_dist_item = self._prepare_dependency_progress()
+        return self._dependency_dist_item
 
-    venv_dest_dir = os.path.join(local_venv_dir, package.name)
-    log.debug(f"Checking {package.name} in {local_venv_dir}")
+    def get_dependency_metadata_filepath(self):
+        return os.path.join(self._dependency_dirpath, "dependency.json")
 
-    if os.path.isdir(venv_dest_dir):
-        log.debug(f"Venv folder {venv_dest_dir} already exists.")
-        return
+    def get_addons_metadata_filepath(self):
+        return os.path.join(self._addons_dirpath, "addons.json")
 
-    os.makedirs(venv_dest_dir)
+    def read_metadata_file(self, filepath, default_value=None):
+        """Read json file from path.
+
+        Method creates the file when does not exist with default value.
+
+        Args:
+            filepath (str): Path to json file.
+            default_value (Union[Dict[str, Any], List[Any], None]): Default
+                value if the file is not available (or valid).
+
+        Returns:
+            Union[Dict[str, Any], List[Any]]: Value from file.
+        """
+
+        if not os.path.exists(filepath):
+            dirpath = os.path.dirname(filepath)
+            if not os.path.exists(dirpath):
+                os.makedirs(dirpath)
+
+            if default_value is None:
+                default_value = {}
+            with open(filepath, "w") as stream:
+                json.dump(default_value, stream)
```

Comment:
Wait, wait... why is this writing a file in a function called `read_metadata_file`? The docstring mentions it but boy, that's a confusing method name?

Why does it have to generate the file?